### PR TITLE
Fix: Missing Customer Address

### PR DESCRIPTION
**Pull request by rithiksharma100**

### CODE REVIEW

File: includes/helpers/class-simpl-wc-helper.php [hunk 1]
```diff
@@ -223,6 +223,24 @@ static protected function simpl_create_new_customer($order) {
...
     }
 
+    static function simpl_set_customer_address_from_order($customer, $order) {
+        $billing_address = $order->get_address('billing');
+        $shipping_address = $order->get_address('shipping');
+
+        if(isset($shipping_address) && isset($billing_address)) {        
+            foreach($shipping_address as $key => $value) {
+                if(method_exists(WC()->customer, "set_shipping_".$key)) {
+                    WC()->customer->{"set_shipping_".$key}($value);
```
Comment (piyush-getsimpl) [inline includes/helpers/class-simpl-wc-helper.php:233]: Shouldn't this be applied on the $customer object ?
$customer->{"set_shipping_".$key}($value);

WC()->customer-> may be adding it to the currently logged-in user ?

File: includes/helpers/class-simpl-wc-helper.php [hunk 2]
```diff
@@ -223,6 +223,24 @@ static protected function simpl_create_new_customer($order) {
...
+
+        if(isset($shipping_address) && isset($billing_address)) {        
+            foreach($shipping_address as $key => $value) {
+                if(method_exists(WC()->customer, "set_shipping_".$key)) {
+                    WC()->customer->{"set_shipping_".$key}($value);   
+                }
+            }
+            foreach($billing_address as $key => $value) {
+                if(method_exists(WC()->customer, "set_billing_".$key)) {
+                    WC()->customer->{"set_billing_".$key}($value);
```
Comment (piyush-getsimpl) [inline includes/helpers/class-simpl-wc-helper.php:238]: Same as shipping.

File: includes/helpers/class-simpl-wc-helper.php [hunk 3]
```diff
@@ -223,6 +223,24 @@ static protected function simpl_create_new_customer($order) {
         return $customer;
     }
 
+    static function simpl_set_customer_address_from_order($customer, $order) {
+        $billing_address = $order->get_address('billing');
+        $shipping_address = $order->get_address('shipping');
+
+        if(isset($shipping_address) && isset($billing_address)) {
```
Comment (piyush-getsimpl) [inline includes/helpers/class-simpl-wc-helper.php:230]: Is there a possibility that either of shipping vs billing exists but not the other?
If that is the case, we must not check for both in the same condition with &.
WDYT ?

File: includes/helpers/class-simpl-wc-helper.php (hunk 1 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php:233]: oh yes, done. It was anyways pointing to WC()->customer

File: includes/helpers/class-simpl-wc-helper.php (hunk 3 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php:230]: we send both as same from checkout-3pp, if either isn't present then there is something wrong with the checkout flow itself

Comment (piyush-getsimpl) [reply includes/helpers/class-simpl-wc-helper.php:230]: Cool.

**Review by piyush-getsimpl — DISMISSED** [review]

**Review by piyush-getsimpl — APPROVED** [review]

File: includes/helpers/class-simpl-wc-helper.php [hunk 4]
```diff
@@ -223,6 +223,24 @@ static protected function simpl_create_new_customer($order) {
         return $customer;
     }
 
+    static function simpl_set_customer_address_from_order($customer, $order) {
```
Comment (Chhekur) [inline includes/helpers/class-simpl-wc-helper.php:226]: looks good, just a comment that currently only populates address of new created customer, not for existing customers even if address is not exists.

**Review by Chhekur — APPROVED** [review]

File: includes/helpers/class-simpl-wc-helper.php (hunk 4 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php:226]: yeah

Comment (Chhekur) [reply includes/helpers/class-simpl-wc-helper.php:226]: I meant ki let's add a comment in codebase to clarify! but it's fine.

**Pull request merged**